### ISSUE TITLE
Docker Desktop 4.12.0 windows installation fails

### ISSUE THREAD

**Issue by abirstolov** [docker/for-win#12994]
Unable to get the Docker Desktop for windows installed.

I've tried:
1. Starting installer interactively
2. Starting installer via Power-shell (Admin)

  - [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics

### Actual behavior
Getting 'Installation canceled' in the GUI

### Expected behavior
Successful installation

### Information
  - Is it reproducible? Yes
  - Windows Version: 10
  - Docker Desktop Version: 4.12.0
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Steps to reproduce the behavior

Attached Console log:
```
Version: 4.12.0 (85629)
Sha1:
Started on: 2022/10/03 09:31:34.526
Resources: C:\Users\Abir\Desktop\resources
OS: Windows 10 Pro
Edition: Professional
Id: 2009
Build: 19044
BuildLabName: 19041.1.amd64fre.vb_release.191206-1406
File: C:\ProgramData\DockerDesktop\install-log-admin.txt
CommandLine: "C:\Users\Abir\Desktop\docker.exe" install
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[2022-10-03T09:31:34.745196000Z][ManifestAndExistingInstallationLoader][Info   ] No install path specified, looking for default installation registry key
[2022-10-03T09:31:34.760823500Z][Installer         ][Info   ] No installation found
[2022-10-03T09:31:34.990261900Z][InstallWorkflow   ][Info   ] Cancel pending background download
[2022-10-03T09:31:34.990261900Z][BackgroundTransfer][Info   ] Cancel current background transfer job
[2022-10-03T09:31:35.005886000Z][InstallWorkflow   ][Info   ] Using package: res:DockerDesktop
[2022-10-03T09:31:35.005886000Z][InstallWorkflow   ][Info   ] Downloading
[2022-10-03T09:31:36.772819100Z][InstallWorkflow   ][Info   ] Extracting manifest
[2022-10-03T09:31:37.537689900Z][InstallWorkflow   ][Info   ] Manifest found: version=85629, displayVersion=4.12.0, channelUrl=https://desktop.docker.com/win/main/amd64/appcast.xml
[2022-10-03T09:31:37.537689900Z][InstallWorkflow   ][Info   ] Checking prerequisites
[2022-10-03T09:31:37.988696100Z][InstallWorkflow   ][Info   ] Msi based installation found, prompting user
[2022-10-03T09:31:38.010620100Z][Installer         ][Warning] Failed to track the installer started event
[2022-10-03T09:31:40.140057800Z][InstallWorkflow   ][Info   ] Uninstalling msi based installation, with special flag
[2022-10-03T09:31:40.226684800Z][InstallWorkflow   ][Error  ] MSI Uninstall failed
[2022-10-03T09:31:40.229684900Z][InstallWorkflow   ][Info   ] Rollbacking component CommunityInstaller.ExtractManifestStep
[2022-10-03T09:31:40.231684000Z][InstallWorkflow   ][Info   ] Rollbacking component CommunityInstaller.DownloadStep
```

**Comment by CyberCyclistMSN** [comment]
This worked for me.  If fixes corrupted WMI repositories.

Put this script into a .bat file.  Run it as administrator.
Restart Windows
Reinstall Docker

echo on
cd /d c:\temp
if not exist %windir%\system32\wbem goto TryInstall
cd /d %windir%\system32\wbem
net stop winmgmt
winmgmt /kill
if exist Rep_bak rd Rep_bak /s /q
rename Repository Rep_bak
for %%i in (*.dll) do RegSvr32 -s %%i
for %%i in (*.exe) do call :FixSrv %%i
for %%i in (*.mof,$.mfl) do Mofcomp -s %%i
net start winmgmt
goto End

:FixSrv
if /I (%1) == (wbemcntl.exe) goto SkipSrv
if /I (%1) == (wbemtest.exe) goto SkipSrv
if /I (%1) == (mofcomp.exe) goto SkipSrv

:SkipSrv
goto End

:TryInstall
if not exist wmicore.exe goto End
wmicore /s
net start winmgmt

:End

**Comment by flvrm92** [comment]
I had the same problem as you after trying to update from 4.11.1 to 4.12.0. I noticed my docker was removed and not installed again, so I tried to install it manually, but without success. I found some tutorials to check my [WMI repositories health](https://kb.acronis.com/content/62731) (basically, what @CyberCyclistMSN recommended), but it was consistent. What really solved my problem was to run a [chkdsk](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/chkdsk?tabs=event-viewer) /R to check and fix my os partition.
